### PR TITLE
Implemented JWT authentication for HTTP API, refactored Axum app state

### PR DESCRIPTION
**Pull request by spetz**

### CODE REVIEW

File: server/src/http/claims.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JwtClaims {
```
Comment (hubcio) [inline server/src/http/claims.rs:4]: rest of the structs in this file starts with Http, perhaps this one should too?
`HttpJwtClaims`?

File: server/src/http/jwt.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    State(state): State<Arc<AppState>>,
+    mut request: Request<T>,
+    next: Next<T>,
+) -> Result<Response, StatusCode> {
+    match request.uri().path() {
+        "/" => return Ok(next.run(request).await),
+        "/ping" => return Ok(next.run(request).await),
+        "/users/login" => return Ok(next.run(request).await),
+        _ => (),
+    }
```
Comment (hubcio) [inline server/src/http/jwt.rs:31]: if you define these on top:
```rust
const UNAUTHORIZED: StatusCode = StatusCode::UNAUTHORIZED;
const AUTHORIZATION: &str = "Authorization";
const BEARER: &str = "Bearer ";
```

you can use them later as:
```rust
    let header = request
        .headers()
        .get(AUTHORIZATION)
        .ok_or(UNAUTHORIZED)?;
```
Looks a bit cleaner.

File: server/src/http/jwt.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    pub user_id: u32,
+}
+
+pub async fn jwt_auth<T>(
+    State(state): State<Arc<AppState>>,
+    mut request: Request<T>,
+    next: Next<T>,
+) -> Result<Response, StatusCode> {
+    match request.uri().path() {
+        "/" => return Ok(next.run(request).await),
```
Comment (hubcio) [inline server/src/http/jwt.rs]: define helper function:
```rust
fn should_skip_auth(path: &str) -> bool {
    matches!(path, "/" | "/ping" | "/users/login")
}
```

and use it:
```rust
    if should_skip_auth(request.uri().path()) {
        return Ok(next.run(request).await);
    }
```

File: server/src/http/claims.rs (hunk 1 above)
Comment (spetz) [reply server/src/http/claims.rs:4]: I'd stick to JWT as its an acronym on its own :)

**Review by hubcio — APPROVED** [review]

**Pull request merged**